gateway-api/src/lib.rs

```diff
@@ -87,29 +88,30 @@ mod tests {
         assert!(gw.metadata.name.is_some());
         assert!(gw.metadata.uid.is_some());
 
-        let mut gw_status = GatewayStatus::default();
-        gw_status.addresses = Some(vec![GatewayStatusAddresses::default()]);
-        gw_status.listeners = Some(vec![GatewayStatusListeners {
-            name: "tcp".into(),
-            attached_routes: 0,
-            supported_kinds: vec![],
-            conditions: vec![Condition {
+        let gw_status = GatewayStatus {
+            addresses: Some(vec![GatewayAddress::default()]),
```

Comment:
nit: you could use the `..default::Default()` struct update syntax at the bottom instead of initializing this as default.

Reply:
Done

Reply:
Actually, I had to revert this one.. it broke the integration tests.